File: packages/schemas/src/signal.ts
import { z } from "zod";
import { Id } from "./primitives.js";
import { SignalConfidence } from "./confidence.js";

export const Sentiment = z.enum(["positive", "negative", "neutral", "mixed"]);
export type Sentiment = z.infer<typeof Sentiment>;

export const Signal = z.object({
  id: Id,
  topic: z.string().min(1),
  sentiment: Sentiment,
  claim: z.string().min(1),
  evidenceIds: z.array(Id).min(1),
  confidence: SignalConfidence,
});
export type Signal = z.infer<typeof Signal>;

File: backend/src/main/java/com/acltabontabon/kuro/domain/Signal.java
package com.acltabontabon.kuro.domain;

import java.util.List;

// mirrors packages/schemas/src/signal.ts
public record Signal(
        String id,
        String topic,
        Sentiment sentiment,
        String claim,
        List<String> evidenceIds,
        SignalConfidence confidence) {

    public Signal {
        evidenceIds = List.copyOf(evidenceIds);
    }
}
